fix: don't show fabulous on unsupported systems

contariaa opened the pull request:
Vanilla hides "Fabulous!" on unsupported GPU's, this simply copies the check used by vanilla.
Works correctly even when the value is set to fabulous through options.txt, the sodium menu will simply display "Fabulous!", then when pressed skip to fancy -> fast -> fancy -> fast -> ...

contariaa committed:
fix: don't show fabulous on unsupported systems

jellysquid3 commented on common/src/main/java/net/caffeinemc/mods/sodium/client/gui/options/control/CyclingControl.java:
```diff
@@ -50,6 +50,12 @@ public CyclingControl(Option<T> option, Class<T> enumType, T[] allowedValues) {
         }
     }
 
+    public CyclingControl(Option<T> option, T[] allowedValues, Component[] names) {
```
There should be validation that `allowedValues` and `names` are the same length, and that `allowedValues` is an array of the correct type.

contariaa replied:
In the current implementation  `allowedValues` and `names` are not guaranteed to be of the same length, the only requirement is that `allowedValues.length <= names.length`. Specifically `allowedValues` is an array of length 3 (default) or 2 (on systems that do not support fabulous graphics), while `names` is always an array of length 3 containing all the translations.
Would you like me to validate that `allowedValues.length <= names.length` or change the passed `names` to always match the length?

contariaa replied:
Also, validation seems to only be present in one out of the two already existing constructors, would you like me to add that in this PR?

contariaa replied:
Taking another look at it, the existing constructor `public CyclingControl(Option<T> option, Class<T> enumType, T[] allowedValues)` also initializes `names` with the length of the `universe` (`enumType.getEnumConstants()`) rather than the length of `allowedValues`.
In `CyclingControl.CyclingControlElement#render`, the `names` array is queried not with the index of the value in `allowedValues`, but rather the values enum ordinal, so it seems appropriate to validate `names` to always match the length of the enums universe, not just its allowed values.

contariaa replied:
I've pushed validation that makes sense to me, `names` should always be the same length as the entire enum universe because its accessed with the enum ordinal, not the index of the value in the `allowedValues` array

contariaa committed:
fix: add validation for CyclingControl allowedValues

contariaa committed:
fix: validate against the given enumType when possible
I originally copied this from the constructor that isn't passed the enum type, however it makes more sense to compare against it than just the elements to one another
